### PR TITLE
feat(trading): clarify collateral and breakdown tables

### LINKED ISSUE #2871: Clarify collateral and breakdown tables
## The Chore

We need to make better sense of the way we display collateral, used margin, total margin available so that people understand 
* How much they have left 
* What is using the most margin of the markets they hold positions in 

## Tasks
- [x] Get to grips with what we currently have and where the issues are (session with Michael, Matt?) 
- [x] Look at how this is handled most commonly by the biggest players - DYDX and Binance
- [x] Sketch out an information architecture that addresses the main needs coming from the manual testing remembering to consider how this impacts what we call collateral / assets in the wallet side panel on governance 
- [x] Agree improvements to be made 
- [ ] Implement 
- [ ] Test 

#### In collateral table
- [ ] Rename 'Total' column to 'General'
- [ ] Add a 'Total margin' column and display the sum of all margin accounts for that asset
- [ ] Remove 'Used' column
- [ ] Add 'Fees earned' column and display the sum of all fee accounts (infrastructure, maker, liquidity) for that asset
- [ ] Add 'Bond' column and display the bond account type balance

#### In breakdown table
- [ ] Show full list of un-aggregated accounts for asset
- [ ] Show columns for type, market (if margin account), balance

## Sketch

From [the Miro artboard](https://miro.com/app/board/uXjVPUenJos=/?moveToWidget=3458764547506685987&cot=14)

### Collateral table
Image

"Used" tooltip:
> Currently allocated to a market as margin or bond. Check the breakdown for details.

"Available" tooltip:
> Deposited on the network, but not allocated to a market. Free for use in new markets or orders.

"Total" tooltip:
> The full amount associated with this key. Total of used and available collateral.

### Breakdown modal
Image

## Additional details / background info

https://vegaprotocol.slack.com/archives/CFXA4EGN7/p1675417313628039

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #2871 

# Description ℹ️

Clarify collateral and breakdown tables.

# Demo 📺
Screenshot 2023-03-08 at 12 55 25

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: apps/trading-e2e/src/integration/trading-accounts.cy.ts [hunk 1]
```diff
@@ -22,22 +22,22 @@ describe('accounts', { tags: '@smoke' }, () => {
 
     cy.getByTestId('tab-accounts')
       .get(tradingAccountRowId)
-      .find('[col-id="breakdown"] [data-testid="breakdown"]')
-      .should('have.text', 'Breakdown');
+      .find('[data-testid="accounts-actions"]')
```
Comment (MadalinaRaicu) [inline apps/trading-e2e/src/integration/trading-accounts.cy.ts:25]:
```suggestion
      .find('[col-id="accounts-actions"]')
```

**Review by edd — APPROVED** [review]
Screenshot 2023-03-08 at 17 04 32

Tested as a user, and with 'View as...' for a liquidity provider on an active market. It's working nicely.

File: libs/accounts/src/lib/accounts-table.tsx [hunk 2]
```diff
@@ -76,6 +81,22 @@ export const AccountTable = forwardRef<AgGridReact, AccountTableProps>(
             resizable: true,
             tooltipComponent: TooltipCellComponent,
             sortable: true,
+            comparator: (
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx:93]: asset decimals can be different for different asset accounts. you can't compare numbers that are not formatted

File: libs/accounts/src/lib/accounts-table.tsx [hunk 3]
```diff
@@ -76,6 +81,22 @@ export const AccountTable = forwardRef<AgGridReact, AccountTableProps>(
...
+              valueA: string,
+              valueB: string,
+              nodeA: RowNode,
+              nodeB: RowNode
+            ) => {
+              if (isNumeric(valueA) && isNumeric(valueB)) {
+                const a = toBigNum(valueA, nodeA.data.asset?.decimals);
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx:100]: asset decimals can be different for different asset accounts. you can't compare numbers that are not formatted

File: libs/accounts/src/lib/accounts-table.tsx [hunk 4]
```diff
@@ -111,168 +132,197 @@ export const AccountTable = forwardRef<AgGridReact, AccountTableProps>(
...
+                  <ButtonLink
+                    data-testid="breakdown"
+                    onClick={() => {
+                      setOpenBreakdown(!openBreakdown);
+                      setRow(data);
+                    }}
+                  >
+                    <span className="">
+                      {' '}
+                      {data &&
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx]: updated!

File: libs/accounts/src/lib/accounts-table.tsx [hunk 5]
```diff
@@ -186,115 +204,113 @@ export const AccountTable = forwardRef<AgGridReact, AccountTableProps>(
             field="available"
             type="rightAligned"
             headerTooltip={t(
-              'This is the amount of collateral available in your general account.'
+              'Deposited on the network, but not allocated to a market. Free for use in new markets or orders.'
```
Comment (candida-d) [inline libs/accounts/src/lib/accounts-table.tsx:207]: What does "new markets" want to imply here? (And why is it contrasted to "orders"?)

File: libs/accounts/src/lib/accounts-table.tsx [hunk 6]
```diff
@@ -186,115 +204,113 @@ export const AccountTable = forwardRef<AgGridReact, AccountTableProps>(
...
-            colId="breakdown"
-            headerName=""
-            sortable={false}
-            minWidth={200}
+            headerName={t('Total')}
             type="rightAligned"
-            cellRenderer={({
+            field="total"
+            headerTooltip={t(
+              'The full amount associated with this key. Total of used and available collateral.'
```
Comment (candida-d) [inline libs/accounts/src/lib/accounts-table.tsx:227]: Associated is a specific status for VEGA tokens in order to stake - is it the same meaning here?

File: libs/accounts/src/lib/accounts-table.tsx (hunk 5 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx:207]: @neildawson here are some suggestions for the tooltips.

File: libs/accounts/src/lib/accounts-table.tsx (hunk 6 above)
Comment (candida-d) [inline libs/accounts/src/lib/accounts-table.tsx:227]:
```suggestion
              'The full amount of assets on this key. Total of used and available collateral.'
```

**Review by candida-d — APPROVED** [review]

**Pull request merged**